catalog/main.go

```diff
@@ -47,3 +49,77 @@ func readFile(path string) []byte {
 	}
 	return result
 }
+
+// Recommend function creates instance of catalog with recommendations for installation
+func Recommend(projectPath string) Catalog {
+	var result Catalog
+	extensions := make(map[string]bool)
+	var temp[] string
+	filepath.Walk(projectPath, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			temp = strings.SplitAfter(path, ".")
+			extensions["."+temp[len(temp)-1]] = true
+		}
+		return nil
+	})
+	//get list of needable languages
+	langs := make(map[string]bool)
+	dataLang, e := ioutil.ReadFile(catalogFolder+"/language.linguist.json")
+	if e != nil {
+		fmt.Printf("Catch error while reading file: %v\n", e)
+		os.Exit(1)
```

Comment:
I suggest to avoid such constructs (see [`Dont Use Exceptions For Flow Control`](http://wiki.c2.com/?DontUseExceptionsForFlowControl)).
Please also note that adding logging to this code will be pretty complex - there is no single place where I can handle exceptions from this function and log it to file, so I suggest to move exception handling at least one level upper and avoid `os.Exit` if possible.